### PR TITLE
os/drivers : Enable KERNEL_TEST_DRV automatically when test case enabled

### PR DESCRIPTION
**Pull request by YiHyunJin**
test_drv is need at kernel test.
So, when CONFIG_EXAMPLES_KERNEL_SAMPLE enable,
it makes KERNEL_TEST_DRV enable.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [7b66044da6c56a00230df9fd1f9ed3f6d0e6f0a5] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [7b66044da6c56a00230df9fd1f9ed3f6d0e6f0a5] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@YiHyunJin Please write a title what you did exactly.

**Comment by sunghan-chang** [comment]
@YiHyunJin Please split commits for different kinds of changes.

**Comment by sunghan-chang** [comment]
@xiarihf Please find this. This PR covers your PR, #2658 .

**Comment by seinfra** [comment]
Target : [e40562a5ef4613c605115786e2d796220d804609] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e40562a5ef4613c605115786e2d796220d804609] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@YiHyunJin Please resolve a conflict.

File: os/drivers/Kconfig [hunk 1]
```diff
@@ -481,4 +481,7 @@ menuconfig DRIVERS_WIRELESS
 		Drivers for various wireless devices.
 
 source drivers/wireless/Kconfig
+
+comment "Kernel test case Device Options"
```
[the remark this answers is not in the document]
Comment (jeongchanKim) [reply os/drivers/Kconfig]: I think it looks like that this config is related with wireless. for avoiding this, you'd better move this config to above

File: os/drivers/testcase/kernel_testcase_drv.c [hunk 2]
```diff
@@ -196,14 +196,14 @@ static ssize_t testdrv_write(FAR struct file *filep, FAR const char *buffer, siz
...
  * Description:
  *   Register /dev/testcase
  *
  ****************************************************************************/
 
-void test_drv_register(void)
+void kernel_test_drv_register(void)
 {
-	(void)register_driver(TESTCASE_DRVPATH, &testdrv_fops, 0666, NULL);
+	(void)register_driver(TESTCASE_DRVPATH, &kernel_test_drv_fops, 0666, NULL);
```
Comment (jeongchanKim) [inline os/drivers/testcase/kernel_testcase_drv.c]: How about changing TESTCASE_DRVPATH -> KERNEL_TC_DRVPATH?

Comment (YiHyunJin) [reply os/drivers/testcase/kernel_testcase_drv.c]: Yes. I thinks so.

File: os/drivers/Kconfig (hunk 1 above)
[the remark this answers is not in the document]
Comment (thapav) [reply os/drivers/Kconfig]: I think renaming this comment should do it. You could remove options from the title.

**Comment by seinfra** [comment]
Target : [88da4d4dbbca9c50548d2d9030e8fc994a6e4f8d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [88da4d4dbbca9c50548d2d9030e8fc994a6e4f8d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e502bfb4eba1ded0f588e63289c95633fbec18e5] -

**Comment by seinfra** [comment]
Target : [e502bfb4eba1ded0f588e63289c95633fbec18e5] -

**Comment by seinfra** [comment]
Target : [bd1a1a36d366b9356638942fbad2f68e5fa288c6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bd1a1a36d366b9356638942fbad2f68e5fa288c6] - Code Rule Check (C++) OK.

File: build/configs/cy4390x/cy4390x_demo/defconfig [hunk 3]
```diff
@@ -482,12 +482,12 @@ CONFIG_UART2_2STOP=0
 # T-trace
 #
 # CONFIG_TTRACE is not set
+# CONFIG_KERNEL_TEST_DRV is not set
```
Comment (sunghan-chang) [inline build/configs/cy4390x/cy4390x_demo/defconfig]: Let's locate this next of `CONFIG_POWER`

**Comment by seinfra** [comment]
Target : [4397ee66ed2dabc45a200ffee9761b5171e3ee13] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4397ee66ed2dabc45a200ffee9761b5171e3ee13] - Code Rule Check (C++) OK.

File: os/drivers/Kconfig [hunk 4]
```diff
@@ -428,6 +428,8 @@ menuconfig POWER
 		Enable building of power-related devices (battery monitors, chargers,
 		etc).
 
+source drivers/testcase/Kconfig
```
Comment (sunghan-chang) [inline os/drivers/Kconfig]: Wrong location. Let's move another location.

**Comment by seinfra** [comment]
Target : [057d38b514f689b7d3bfbe0c7640704aa5239880] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [057d38b514f689b7d3bfbe0c7640704aa5239880] - Code Rule Check OK.

File: build/configs/artik053/tc/defconfig [hunk 5]
```diff
@@ -437,6 +437,7 @@ CONFIG_NETDEV_MULTINIC=y
 CONFIG_PIPES=y
 CONFIG_DEV_PIPE_SIZE=1024
 CONFIG_POWER=y
+CONFIG_KERNEL_TEST_DRV=y
```
Comment (sunghan-chang) [inline build/configs/artik053/tc/defconfig]: not modified here.

File: build/configs/qemu/build_test/defconfig [hunk 6]
```diff
@@ -447,6 +447,7 @@ CONFIG_WATCHDOG=y
 CONFIG_WATCHDOG_DEVPATH="/dev/watchdog0"
 CONFIG_TIMER=y
 CONFIG_ANALOG=y
+CONFIG_KERNEL_TEST_DRV=y
```
Comment (sunghan-chang) [inline build/configs/qemu/build_test/defconfig]: check here.

File: build/configs/sidk_s5jt200/tc/defconfig [hunk 7]
```diff
@@ -413,6 +413,7 @@ CONFIG_WATCHDOG=y
 CONFIG_WATCHDOG_DEVPATH="/dev/watchdog0"
 # CONFIG_TIMER is not set
 CONFIG_ANALOG=y
+CONFIG_KERNEL_TEST_DRV=y
```
Comment (sunghan-chang) [inline build/configs/sidk_s5jt200/tc/defconfig]: here also

**Comment by seinfra** [comment]
Target : [840ddef1121f1425745e23e5c0d2683ec0dd963c] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [840ddef1121f1425745e23e5c0d2683ec0dd963c] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b48f74a05bfc0b37efbb6e266068f7f01508f581] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [b48f74a05bfc0b37efbb6e266068f7f01508f581] - Code Rule Check OK.

**Pull request merged**